lib/libspl/Makefile.am

```diff
@@ -17,7 +17,10 @@ AM_CFLAGS += $(LIBTIRPC_CFLAGS)
 AM_CCASFLAGS = \
 	$(CFLAGS)
 
-noinst_LTLIBRARIES = libspl.la
+noinst_LTLIBRARIES = libspl_assert.la libspl.la
```

Comment:
It is to allow it to be linked into libnvpair. libnvpair doesn't use anything else from libspl so it seems like a waste to link it all in. libspl is still duplicated in libuutil and libzfs_core, though.

The previous version, which broke when building in the source directory, tried to link in just assert.c directly, but that doesn't work because make finds assert.lo in the VPATH (built during libspl compilation) and thinks it's up-to-date and doesn't need to recompiled for libnvpair, but libtool can't find the object file because it only looks in the libnvpair subdirectory. Building it into a convenience library fixes this issue.